### ISSUE TITLE
Add tunnel CONNECT response headers to httplib / http.client

### ISSUE THREAD

**Issue by 0fa05392-c1fa-4949-9e9e-f6afd213499a** [python/cpython#69152]
BPO | [24964](https://bugs.python.org/issue24964)
--- | :---
Nosy | @terryjreedy, @vadmium, @OneMoreZanuda
PRs | <li>python/cpython#26152</li>
Files | <li>[httplib.py](https://bugs.python.org/file40301/httplib.py "Uploaded as text/plain at 2015-08-30.17:41:46 by thomas.belhalfaoui"): Proposed patch for httplib.py</li><li>[http-detach.patch](https://bugs.python.org/file40371/http-detach.patch "Uploaded as text/plain at 2015-09-05.06:27:33 by @vadmium"): HTTPConnection.detach() implementation only</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2015-08-30.17:41:50.140>
labels = ['type-feature', 'library', '3.11']
title = 'Add tunnel CONNECT response headers to httplib / http.client'
updated_at = <Date 2021-05-15.21:19:25.060>
user = 'https://bugs.python.org/thomasbelhalfaoui'
```

bugs.python.org fields:
```python
activity = <Date 2021-05-15.21:19:25.060>
actor = 'alexey.namyotkin'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2015-08-30.17:41:50.140>
creator = 'thomas.belhalfaoui'
dependencies = []
files = ['40301', '40371']
hgrepos = []
issue_num = 24964
keywords = ['patch']
message_count = 10.0
messages = ['249361', '249373', '249741', '249803', '249897', '249906', '249996', '250085', '393729', '393730']
nosy_count = 4.0
nosy_names = ['terry.reedy', 'martin.panter', 'thomas.belhalfaoui', 'alexey.namyotkin']
pr_nums = ['26152']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue24964'
versions = ['Python 3.11']
```

</p></details>

### Linked PRs
* gh-104248

**Comment by gpshead** [comment]
thanks for the contribution!
